File: 2.hs
fib :: Int -> Int
fib 0 = 0
fib 1 = 1
fib x = fib (x - 1) + fib (x - 2)

--fibs :: Int -> [Int]
--fibs 1 = [1]
--fibs 2 = [2, 1]
--fibs x = ((head (fibs (x - 1))) + (head (fibs (x - 2)))):fibs (x - 1)

fibs' :: [Int] -> [Int]
fibs' [] = [1]
fibs' [1] = [2, 1]
fibs' all@(x1:x2:xs) = (x1 + x2):all    

fibs max = takeWhile

evenFibSum :: Int -> Int
evenFibSum upper = sum [x | x <- map fib [1..], x < upper, even x]

File: 2.swift
func fib(i: Int) -> Int {
    switch i {
    case 1, 2: return i
    default:
        var sum = 0
        var (first, second) = (1, 2)
        for j in 3...i {
            sum = first + second
			first = second
            second = sum
        }
        return sum
    }
}

func evensum(max: Int) -> Int {
    var i = 1
    var sum = 0
    var value = 0
    while value < max {
        value = fib(i)
        if value % 2 == 0 {
            sum += value
        }
        i += 1
    }
    return sum
}

// println(evensum(Int(4e6)))